File: app/src/main/java/com/MohamedTaha/Imagine/New/Adapter/AdapterForAzkar.java
package com.MohamedTaha.Imagine.New.Adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.MohamedTaha.Imagine.New.R;
import com.MohamedTaha.Imagine.New.mvp.model.ModelAzkar;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class AdapterForAzkar extends RecyclerView.Adapter<AdapterForAzkar.RecyclerAzkarViewHolder> {
    private List<ModelAzkar> modelAzkars = new ArrayList<>();
    private ClickListener clickListener;

    public AdapterForAzkar(List<ModelAzkar> modelAzkars, ClickListener clickListener) {
        this.modelAzkars = modelAzkars;
        this.clickListener = clickListener;
    }

    @NonNull
    @Override
    public RecyclerAzkarViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.custom_azkar, parent, false);
        RecyclerAzkarViewHolder holder = new RecyclerAzkarViewHolder(view);
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                int position = holder.getAdapterPosition();
                if (clickListener != null) clickListener.onClick(view, position);
            }
        });
        return holder;
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerAzkarViewHolder holder, int position) {
        ModelAzkar azkar = modelAzkars.get(position);
        holder.TVNameAzkar.setText(azkar.getName_azkar());

    }

    @Override
    public int getItemCount() {
        return modelAzkars.size();
    }

    public interface ClickListener {
        void onClick(View view, int position);
    }

    public static class RecyclerAzkarViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.TV_Name_Azkar)
        TextView TVNameAzkar;
        private View view;

        public RecyclerAzkarViewHolder(@NonNull View itemView) {
            super(itemView);
            view = itemView;
            ButterKnife.bind(this, view);
        }
    }
}

File: app/src/main/java/com/mohamedtaha/imagine/ui/home/adapter/AdapterForAzkar.kt
package com.mohamedtaha.imagine.ui.home.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.mohamedtaha.imagine.databinding.CustomAzkarBinding
import com.mohamedtaha.imagine.mvp.model.ModelAzkar
import com.mohamedtaha.imagine.util.ClickListener

class AdapterForAzkar(private val clickListener: ClickListener<Int>) :
    RecyclerView.Adapter<AdapterForAzkar.AzkarViewHolder>() {
    private lateinit var modelAzkars: List<ModelAzkar>
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): AzkarViewHolder {
        val viewHolder = AzkarViewHolder(
            CustomAzkarBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        )
        viewHolder.itemView.setOnClickListener {
            val position = viewHolder.adapterPosition
            clickListener.onClick(it, position)
        }
        return viewHolder
    }

    override fun onBindViewHolder(holder: AzkarViewHolder, position: Int) {
        val azkar = modelAzkars[position]
        holder.bind(azkar)
    }

    override fun getItemCount(): Int {
        return modelAzkars.size
    }
    fun setData(modelAzkar: List<ModelAzkar>){
        this.modelAzkars = modelAzkar
    }

    class AzkarViewHolder(val binding: CustomAzkarBinding) : RecyclerView.ViewHolder(binding.root){
        fun bind(modelAzkar: ModelAzkar){
            binding.modelAzkar = modelAzkar
            binding.executePendingBindings()
        }
    }
}